src/main/java/org/jenkinsci/test/acceptance/controller/WinstoneController.java

```diff
@@ -26,13 +28,23 @@
 
     private static final List<String> JENKINS_JAVA_OPTS = envVarOpts("JENKINS_JAVA_OPTS");
     private static final List<String> JENKINS_OPTS = envVarOpts("JENKINS_OPTS");
+    private final List<String> SYSTEM_PROPERTIES = new ArrayList<>();
 
     private static List<String> envVarOpts(String jenkins_opts) {
         String getenv = System.getenv(jenkins_opts);
         if (getenv == null) return Collections.emptyList();
         return Arrays.asList(getenv.split("\\s+"));
     }
 
+    public void addSystemProperty(String systemProperty) {
+        if (StringUtils.isNotBlank(systemProperty)) {
+            if (!systemProperty.startsWith("-D")) {
+                systemProperty = "-D" + systemProperty;
+            }
+            SYSTEM_PROPERTIES.add(systemProperty);
```

Comment:
I would not expect that someone really creates a `WithSystemProperty("-DFoobar=wibble").  or calls `addSystemPropety("-Dfoo=bar")` 
Otherwise these are not system properties, they are specific java arguments.  It may also make having a systemproperty starting with `-D` interesting (not even sure that is possible however...)

Reply:
So, IIUC, `SYSTEM_PROPERTIES` should be renamed to `TEST_JAVA_OPT` (and the method as well) and the add "-D" sentence should be removed, so it's the caller who must add the "-D" 

> It may also make having a system property starting with -D interesting (not even sure that is possible however...)

it is possible. I've tested locally and using a break point, I've accessed the Jenkins page to see the System properties and it was there.

Reply:
well that is one route.  :)
the other was to not do the check that it `startsWith` `-D` and add that only at the last minute unconditionally